fcrepo-auth-webac/src/main/java/org/fcrepo/auth/webac/WebACFilter.java

```diff
@@ -106,21 +122,85 @@ public void destroy() {
         // this method intentionally left empty
     }
 
+    private FedoraSession session() {
+        if (session == null) {
+            session = sessionFactory.getInternalSession();
+        }
+        return session;
+    }
+
+    private FedoraResource resource(URI requestURI) {
+        if (resource == null) {
+            resource = nodeService.find(session(), requestURI.toString());
+        }
+        return resource;
+    }
+
+    private String findNearestParent(final String baseURL, final String childPath) {
```

Comment:
This is the helper used most commonly throughout the codebase:
https://github.com/fcrepo4/fcrepo4/blob/master/fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/utils/FedoraTypesUtils.java#L360

Reply:
@awoods In 066b2a1 I have the lookup just being based on strings, and returning a string, since I don't actually need to retrieve the node for some of these checks.